adt_press/nodes/config_nodes.py

```diff
@@ -36,16 +73,74 @@ def custom_plate_path_config(config: DictConfig) -> str:
     return str(config.get("custom_plate_path", ""))
 
 
-def input_language_config(config: DictConfig) -> str:
-    return str(config.get("input_language", "en"))
-
-
-def plate_language_config(config: DictConfig) -> str:
-    return str(config.get("plate_language", "en"))
-
-
-def output_languages_config(config: DictConfig) -> list[str]:
-    return list[str](config["output_languages"])
+def input_language_config(
+    config: DictConfig,
+    language_detection_prompt_config: PromptConfig,
+    pdf_texts: dict[str, PageTexts],
+) -> str:
+    override_value = OmegaConf.select(config, "input_language", default=None)
+    override = _clean_language_override(override_value)
+    if override:
+        log.info("input language override used", language=override)
+        return override
+
+    sample_text = _sample_pdf_text_for_language_detection(pdf_texts)
+    if not sample_text.strip():
+        log.info("input language defaulted to english", reason="no_text")
+        return "en"
+
+    try:
+        response = run_async_task(lambda: detect_input_language(sample_text, language_detection_prompt_config))
+        confidence = getattr(response, "confidence", None)
+        log.info("input language detected automatically", language=response.language_code, confidence=confidence)
+        return response.language_code
+    except Exception as exc:  # pragma: no cover - fallback path
+        log.warning("language detection failed; defaulting to english", error=str(exc))
+        log.info("input language defaulted to english", reason="detection_error")
+        return "en"
```

Comment:
What would cause this exception short of the provider being down? Better to just fail entirely in that case, we aren't likely to succeed going forward anyways.

Reply:
good catch, yes this would probably only get raised if unable to reach the model API